frameworks/DecisionTree/exec.py

```diff
@@ -17,8 +17,8 @@ def run(dataset: Dataset, config: TaskConfig):
 
     is_classification = config.type == 'classification'
 
-    X_train, X_test = impute(dataset.train.X, dataset.test.X)
-    y_train, y_test = dataset.train.y, dataset.test.y
+    X_train, X_test = impute_array(dataset.train.X_enc, dataset.test.X_enc)
```

Comment:
yes, this has always been the case, nothing new in this PR.
In this PR, I started to write some code with the idea to apply a different imputation strategy depending on the feature type, but it was completely out of scope (if I let myself fixing all those issues here, it will never end).
This is worth a separate  PR though.